### PR TITLE
Sle 15 sp4

### PR DESCRIPTION
**Pull request by varkoly**
## Problem

*This module has not been maintained for years and has contained a lot of deprecated and not working stuff.*

- *https://bugzilla.suse.com/show_bug.cgi?id=1207740*
- *https://bugzilla.suse.com/show_bug.cgi?id=1209204*
- *https://bugzilla.suse.com/show_bug.cgi?id=1202112*
- *https://bugzilla.suse.com/show_bug.cgi?id=1211037*

## Solution

Usage SuSEFirewall2 was replaced by firewall-cmd and HANA-firewall
Implemented 3 kind of firewall handling: 
- turn off
- already configured
- configure

Implemented more prechecks (SID, Instant Number, Backup)
Remove checks for older HANA version. Only HANA 2.0 and newer is supported. HANA 1.0 no longer plays a role in new installations.

## Testing

- *Adapted unit test*
- *Tested manually https://documentation.suse.com/sles-sap/15-SP3/html/SLES-SAP-guide/cha-cluster.html*

### CODE REVIEW

**Review by lslezak — COMMENTED** [review]
Some minor comments

File: .yardopts [hunk 1]
```diff
@@ -1,7 +0,0 @@
---no-private
```
[the remark this answers is not in the document]
Comment (varkoly) [reply .yardopts:1]: I did not know what this file is for.

File: src/lib/sap_ha/configuration.rb [hunk 2]
```diff
@@ -224,16 +232,19 @@ def collect_log
...
-    private
+  private
 
     # Load scenarios from the YAML configuration file
     def load_scenarios
       log.debug "--- called #{self.class}.#{__callee__} ---"
-      YAML.load_file(SapHA::Helpers.data_file_path('scenarios.yaml'))
+      begin
+        Psych.unsafe_load_file(SapHA::Helpers.data_file_path("scenarios.yaml"))
+      rescue NoMethodError
```
[the remark this answers is not in the document]
Comment (varkoly) [reply src/lib/sap_ha/configuration.rb:245]: Sorry, I can't understand what you mean by that.

File: src/lib/sap_ha/rpc_server.rb [hunk 3]
```diff
@@ -191,34 +189,23 @@ def immediate_shutdown
...
-      return unless @port_opened
-      rule_no = get_rule_number
-      puts "close_port: rule_no=#{rule_no} #{!!rule_no}"
-      return unless rule_no
-      out, rc = exec_output_status('/usr/sbin/iptables', '-D', 'INPUT', rule_no.to_s)
-      puts "close_port: rc=#{rc}, out=#{out}"
+      _out, status = exec_outerr_status("/usr/bin/firewall-cmd", "--state")
+      return if status.exitstatus != 0
+      out, status = exec_output_status("/usr/bin/firewall-cmd", "--remove-port", "8080/tcp")
+      puts "close_port: status=#{status}, out=#{out}"
```
[the remark this answers is not in the document]
Comment (varkoly) [reply src/lib/sap_ha/rpc_server.rb:206]: Fixed

File: src/lib/sap_ha/sap_ha_unattended_install.rb [hunk 4]
```diff
@@ -42,28 +41,24 @@ def initialize(config)
...
-        log.error e.message
-        NodeLogger.fatal "The imported configuration file did not pass on all checks. Please, review the errors and try again."
-        # Raise the error and let the caller resolve how to present it.
-        raise e
-      end
+      validate_config
+      check_ssh
+      :next
+    rescue UnattendedModeException, ConfigValidationException => e
+      puts e.message
```
[the remark this answers is not in the document]
Comment (varkoly) [reply src/lib/sap_ha/sap_ha_unattended_install.rb:48]: Fixed

File: src/lib/sap_ha/configuration.rb [hunk 5]
```diff
@@ -205,15 +207,21 @@ def dump(slave = false, force = false)
...
       log.debug "--- called #{self.class}.#{__callee__} ---"
-      NodeLogger.info(
-        "Finished setup process on node #{SapHA::NodeLogger.node_name}")
+      NodeLogger.info( "Finished setup process on node #{SapHA::NodeLogger.node_name}")
+      # Start firewall if this was running by starting the module on slave nodes
+      if @fw_state == 0
+        if @role == :master
+	  SapHA::Helpers.write_var_file("need_to_start_firewalld","")
+	else
+          exec_status("/usr/bin/systemctl","start","firewalld")
```
[the remark this answers is not in the document]
Comment (varkoly) [reply src/lib/sap_ha/configuration.rb:222]: Changed

File: src/lib/sap_ha/wizard/hana_page.rb [hunk 6]
```diff
@@ -137,115 +140,132 @@ def hana_backup_popup
...
+          InputField(Id(:hook_db_user_name), Opt(:hstretch), "DB &user name:", "")
+        )
+      end
+
+      def eval_hana_installation
+        return if @my_model.system_id != ""
+        # Read hana installation
+        Dir.glob("/data/SAP_INST/*/installationSuccesfullyFinished.dat") do |f|
+          content = File.read(f)
+          result = content.scan(/SAP HANA System ID:\s+(\w{3}).*SAP HANA Instance:\s+(\w{2})/m)
```
[the remark this answers is not in the document]
Comment (varkoly) [reply src/lib/sap_ha/wizard/hana_page.rb:261]: These files will be created by sap-installation-wizard which is maintained and developed by me :-)
Normaly there can be more then one but mostly there only one. The found values will be used only as suggestion.

[the remark this answers is not in the document]
Comment (lslezak) [reply src/lib/sap_ha/wizard/hana_page.rb:261]: OK, I see. I thought it is generated by some 3rd party tool...

**Pull request merged**